### PR TITLE
Finish switch from Jest to Vitest in API

### LINKED ISSUE #16063: Replace Jest with Vitest
We've been using `vitest` for new projects and it's been an absolute joy. It's more or less a drop-in replacement for `jest` and natively supports ESM. We should replace our current uses of Jest in the sub-packages and API for `vitest`

### PR DESCRIPTION
**Pull request by paescuj**
## Description

Finishing the work started in #16146. Thanks @DorianCBrwn! 🥇
All things are working now.

Since Vitest is ESM first, the following workarounds were necessary. Both can be removed once the API itself has been moved to ESM (#15190):

- Mapping "@directus/format-title" to the version from App via [vitest.config.js](https://github.com/directus/directus/pull/16263/files#diff-7487ea3781cfb53e5b9992fa7a167821bfc8abe58ccefd245681133e10726a95R7-R13)
- ~Use dynamic import in [api/src/database/migrations/run.ts](https://github.com/directus/directus/pull/16263/files#diff-f18d2a3300fdbd8a3d2217edcc3b392ef9b8734700fe8a43d661019ec8b09e2cR66-R67) and [api/src/extensions.ts](https://github.com/directus/directus/pull/16263/files#diff-68995f2e978700b421ed62b416dfdb6e6d97ccdadc7fb535ff9b44640f21ec71R349-R352) when executed via Vitest.
  Not the prettiest solution, but I think it's acceptable, since I expect the changeover to ESM to happen soon.
  (Or someone knows a better solution?)~
  Just realized that the workaround is not even necessary, we can simply replace those `require` statements with `import`. Tested in a Docker container ✅ 

Furthermore, the following global setup is required for the tests to work in environments like the GitHub CI: [globalSetup.js](https://github.com/directus/directus/blob/2293d02cd31fa2e68a71e45ce8611cf49d2c28f8/api/globalSetup.js)

This pull request also contains the following changes:

- Apply default ESLint rules to JS files & treat "vitest.config.js" as a module: [.eslintrc.js](https://github.com/directus/directus/blob/2293d02cd31fa2e68a71e45ce8611cf49d2c28f8/.eslintrc.js)
- Removal of files & packages that are no longer required

Closes #16063, closes #16146.

---

Just in case you were wondering... 🚀

**Jest**:
```console
Test Suites: 27 passed, 27 total
Tests:       163 passed, 163 total
Snapshots:   0 total
Time:        48.212 s
```

**Vitest**:
```console
Test Files  27 passed (27)
     Tests  163 passed (163)
  Start at  15:13:20
  Duration  5.94s (transform 2.39s, setup 5ms, collect 38.89s, tests 723ms)
```

## Type of Change

- [ ] Bugfix
- [ ] Improvement
- [ ] New Feature
- [x] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -221,16 +220,16 @@
 		"@types/uuid": "8.3.4",
 		"@types/uuid-validate": "0.0.1",
 		"@types/wellknown": "0.5.3",
+		"@vitest/ui": "^0.24.5",
```
Comment (paescuj) [inline api/package.json:223]: Might be removed...

**Comment by rijkvanzanten** [comment]
@nickrum Just pinging you seeing there's some minor tweaks to the requiring of code in the extensions loader :

**Comment by paescuj** [comment]
> @nickrum Just pinging you seeing there's some minor tweaks to the requiring of code in the extensions loader :)

Quick note from my side: To my understanding, those dynamic imports are getting transpiled to `require` promises anyway... Also noticed that #15989 will make the `registerHooks` function async as well: https://github.com/directus/directus/blob/d65c38828d643254a6c6eb7042b1675f5d79d06b/api/src/extensions.ts#L339

But I'd be very happy too if @nickrum could take a look at this :-)

File: api/src/services/specifications.test.ts [hunk 2]
```diff
@@ -122,17 +124,17 @@ describe('Integration Tests', () => {
...
+
+						jest
+							.spyOn(CollectionsService.prototype, 'readByQuery')
+							.mockImplementation(jest.fn().mockReturnValue([collection]));
+
+						jest.spyOn(FieldsService.prototype, 'readAll').mockImplementation(
+							jest.fn().mockReturnValue([
+								{
+									collection: collection.collection,
+									field: 'id',
```
Comment (paescuj) [inline api/src/services/specifications.test.ts:137]: Picked wrong side while merging? 😸

Comment (rijkvanzanten) [reply api/src/services/specifications.test.ts]: 🤫

Comment (rijkvanzanten) [reply api/src/services/specifications.test.ts]: All joking aside, no, it was easier to merge it "wrong" to make sure the test was fully up to date with main, and then rework jest->vi after the fact (see commit below :) )

Comment (paescuj) [reply api/src/services/specifications.test.ts]: I see 😃 I'll be quiet and let you do your work 🤫

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**